tobac/utils/internal.py

```diff
@@ -662,3 +662,46 @@ def find_hdim_axes_3D_iris(field_in, vertical_coord=None, vertical_axis=None):
         all_axes[np.logical_not(np.isin(all_axes, [time_axis, vertical_coord_axis]))]
     )
     return output_vals
+
+
+@irispandas_to_xarray
+def detect_latlon_coord_name(in_dataset, latitude_name="auto", longitude_name="auto"):
+    """Function to detect the name of latitude/longitude coordinates
+
+    Parameters
+    ----------
+    in_dataset: iris.cube.Cube, xarray.Dataset, or xarray.Dataarray
+        Input dataset to detect names from
+    latitude_name: str
+        The name of the latitude coordinate. If "auto", tries to auto-detect.
+    longitude_name: str
+        The name of the longitude coordinate. If "auto", tries to auto-detect.
+
+    Returns
+    -------
+    lat_name, lon_name: tuple(str)
+        the detected names of the latitude and longitude coordinates. If
+        coordinate is not detected, returns None for that coordinate.
+
+    """
+    out_lat = None
+    out_lon = None
```

Comment:
I am disinclined from doing so at the moment, because different functions could still proceed without lat/lon being present. One could put in a try/catch block, but this feels cleaner to me. Happy to switch, though.

Reply:
Ah OK, I see now that there are other functions in the process that work fine without lat/lon, so let it be like this! Maybe through a warning or is this not really necessary?

Reply:
I think this isn't necessary, as this should only be used internally, and the various functions that use it can decide to warn/not warn as appropriate.

Reply:
OK, agreed.